### ISSUE TITLE
scan error

### ISSUE THREAD

**Issue by 0x457** [godror/godror#320]
**Describe the bug**
just upgrade from
go1.17 
godror0.29.0

 to
go1.21
godror0.40.3

![image](https://github.com/godror/godror/assets/5813610/247a8fe9-b50d-4c8d-947f-11e283cab11f)

 

**Your oracle client version**
 11g

**Your godror version**
 v0.40.3

**Your go version**
 1.21.3
 

**Machine (please complete the following information):**
 - OS: [e.g. win] mac11.5.2
 - Architecture: [e.g. x86_64]
 - Version: [e.g. 10] go1.21.3 godror0.40.3

**Additional context**

**Comment by tgulacsi** [comment]
What is the error?

**Comment by 0x457** [comment]
The field types in the table have not been modified, so the environment has not been modified except for upgrading  from v0.29 to v0.40.

```sql
sql: Scan error on column index 5, name \"PRODUCTIVITY\": could not convert value '17' to byte array of type 'godror.Number
```

**Comment by tgulacsi** [comment]
Please provide some details: the query, the Go code that does the scan (with the variables' types) and the source table's fields' types.

**Comment by 0x457** [comment]
import "github.com/shopspring/decimal"
struct {
   	Productivity decimal.Decimal `db:"PRODUCTIVITY" json:"productivity"` 
}
------------
oracle's table {
  PRODUCTIVITY number(20,3)
}

v0.29 is ok, upgrade to v0.40 is ng.

**Comment by tgulacsi** [comment]
TL;DR: Use godror.NumberAsString

shopspring/decimal.Decimal implements database/sql.Scanner, but only for float64, int64 and string types.
godror.Number is a string, but Decimal does not handle this, so either you modify Decimal.Scan to handle it (or, even better, implement sql.Composer), or use string instead of godror.Number.

**Comment by 0x457** [comment]
thanks ,
It's a bit difficult for me to implement  sql.Composer, so I have modified the nvarchar type .

**Comment by tgulacsi** [comment]
Just as I said, https://pkg.go.dev/github.com/godror/godror#NumberAsString is just for this: to force the NUMBER columns to be returned as string, not godror.Number.
It's just an option that you have to add to QueryContext or ExecContext as another argument.
